gen/internal/tests/thrift/structs.thrift

```diff
@@ -137,6 +137,14 @@ struct GoTags {
         6: required string FooBarWithRequired (go.tag = 'json:"foobarWithRequired,required"')
 }
 
+struct NotOmitEmpty {
+    1: optional string NotOmitEmptyString (go.tag = 'json:"notOmitEmptyString,!omitempty"')
+    2: optional string NotOmitEmptyInt (go.tag = 'json:"notOmitEmptyInt,!omitempty"')
+    3: optional string NotOmitEmptyBool (go.tag = 'json:"notOmitEmptyBool,!omitempty"')
+    4: optional list<string> NotOmitEmptyList (go.tag = 'json:"notOmitEmptyList,!omitempty"')
+    5: optional map<string, string> NotOmitEmptyMap (go.tag = 'json:"notOmitEmptyMap,!omitempty"')
+}
+
```

Comment:
Can I ask for a case where there is a mix of `!omitempty` and `omitempty`?

Reply:
Sure, will add that in.  The expected behavior will be that !omitempty will take precedence.  !omitempty is an opt-in behavior, whereas because omitempty has become the default in our codegen, it is rarely added, if ever.

Reply:
Added the test cases.  I'm glad you pointed it out because we have to actively strip out `omitempty` when both are present.  Let me know if you have any further comments or questions.